daemon/k8s_watcher.go

```diff
@@ -1444,36 +1452,97 @@ func (d *Daemon) addCiliumNetworkPolicyV1(ciliumV1Store cache.Store, cnp *cilium
 
 	scopedLog.Debug("Adding CiliumNetworkPolicy")
 
+	var rev uint64
+
 	rules, err := cnp.Parse()
 	if err == nil && len(rules) > 0 {
 		d.loadBalancer.K8sMU.Lock()
 		err = k8s.PreprocessRules(rules, d.loadBalancer.K8sEndpoints, d.loadBalancer.K8sServices)
 		d.loadBalancer.K8sMU.Unlock()
 		if err == nil {
-			_, err = d.PolicyAdd(rules, &AddOptions{Replace: true})
+			rev, err = d.PolicyAdd(rules, &AddOptions{Replace: true})
 		}
 	}
 
-	var cnpns cilium_v1.CiliumNetworkPolicyNodeStatus
 	if err != nil {
-		cnpns = cilium_v1.CiliumNetworkPolicyNodeStatus{
-			OK:          false,
-			Error:       fmt.Sprintf("%s", err),
-			LastUpdated: cilium_v1.NewTimestamp(),
-		}
 		scopedLog.WithError(err).Warn("Unable to add CiliumNetworkPolicy")
 	} else {
-		cnpns = cilium_v1.CiliumNetworkPolicyNodeStatus{
-			OK:          true,
-			LastUpdated: cilium_v1.NewTimestamp(),
-		}
 		scopedLog.Info("Imported CiliumNetworkPolicy")
 	}
+	ctrlName := cnp.GetControllerName()
+	cnpCM.UpdateController(ctrlName,
+		controller.ControllerParams{
+			DoFunc: func() error {
+
+				ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+				defer cancel()
+
+				waitForEPsErr := endpointmanager.WaitForEndpointsAtPolicyRev(ctx, rev)
+
+				serverRuleStore, exists, err2 := ciliumV1Store.Get(cnp)
+				if err2 != nil {
+					scopedLog.WithError(err2).Warn("Unable to find v1.CiliumNetworkPolicy in local cache")
+					// Stopping this controller, we have logged the error though
+					return nil
+				}
+				if !exists {
+					// Stopping this controller since the policy doesn't exist.
+					return nil
```

Comment:
Perhaps log a debug here? If this happens rarely it might help with debugging (if it happens a lot then the debug will be annoying)

Reply:
Done